pub/lib/dependabot/pub/package_manager.rb

```diff
@@ -0,0 +1,54 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "dependabot/pub/version"
+require "dependabot/ecosystem"
+require "dependabot/pub/requirement"
+
+module Dependabot
+  module Pub
+    ECOSYSTEM = "dart"
+
+    SUPPORTED_DART_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+    DEPRECATED_DART_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+    class PubPackageManager < Dependabot::Ecosystem::VersionManager
+      extend T::Sig
+
+      NAME = "pub"
+      VERSION = "0.0"
+
+      SUPPORTED_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+      DEPRECATED_VERSIONS = T.let([].freeze, T::Array[Dependabot::Version])
+
+      sig do
+        params(
+          raw_version: String,
+          requirement: T.nilable(Requirement)
```

Comment:
I think you should simplify by removing this require from here and below.

Reply:
@amazimbe , fixed.